### PR TITLE
feat(session-replay): Make session replay support where clause extractor

### PR DESCRIPTION
**Pull request by robbie-c**
## Problem

Session replay doesn't have timestamp clause inlining when using session properties

## Changes

Make the where clause extractor support the raw session replay events table

Screenshot 2024-06-12 at 15 35 24

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Added a test

### CODE REVIEW

**Review by daibhin — APPROVED** [review]

**Comment by pauldambra** [comment]
So the ELI5 is this pushes the timestamp bounds down into the join?

Cool!

**Comment by robbie-c** [comment]
Exactly yeah! Should mean that you aren't querying the whole table, just the specific time range that's being queried.
